internal/reader/onchain_prices_reader.go

```diff
@@ -18,55 +23,58 @@ type TokenPrices interface {
 	GetTokenPricesUSD(ctx context.Context, tokens []ocr2types.Account) ([]*big.Int, error)
 }
 
-type TokenPriceConfig struct {
-	// This is mainly used for inputTokens on testnet to give them a price
-	StaticPrices map[ocr2types.Account]big.Int `json:"staticPrices"`
-}
-
 type OnchainTokenPricesReader struct {
-	TokenPriceConfig TokenPriceConfig
 	// Reader for the chain that will have the token prices on-chain
-	ContractReader commontypes.ContractReader
+	ContractReader commontyps.ContractReader
+	PriceSources   map[types.Account]pluginconfig.ArbitrumPriceSource
+	TokenDecimals  map[types.Account]uint8
 }
 
 func NewOnchainTokenPricesReader(
-	tokenPriceConfig TokenPriceConfig, contractReader commontypes.ContractReader,
+	contractReader commontyps.ContractReader,
+	priceSources map[types.Account]pluginconfig.ArbitrumPriceSource,
+	tokenDecimals map[types.Account]uint8,
 ) *OnchainTokenPricesReader {
 	return &OnchainTokenPricesReader{
-		TokenPriceConfig: tokenPriceConfig,
-		ContractReader:   contractReader,
+		ContractReader: contractReader,
+		PriceSources:   priceSources,
+		TokenDecimals:  tokenDecimals,
 	}
 }
 
+type LatestRoundData struct {
```

Comment:
Can you reference the solidity function in a comment above this struct that returns this information? Make sure to use a permalink